src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const history = [];
+  let currentState = { ...state };
```

Comment:
The initial cloning of `state` into `currentState` is correct. However, ensure that `state` is a plain object. If `state` contains nested objects, a deep clone might be necessary to avoid unintended mutations.